internal/vm/setup.go

```diff
@@ -76,6 +76,28 @@ _G["process_auto_fields"] = function(plugin)
   return json.encode(p)
 end
 
+_G["process_auto_fields_and_transform"] = function(plugin)
+  if type(plugin) ~= "string" then
+    return nil, "invalidate type, expected string"
+  end
+  local tbl, err = json.decode(plugin)
+  if err then
+    return nil, err
+  end
+  local p = Plugins:process_auto_fields(tbl, "insert")
```

Comment:
```suggestion
  local p
  p, err = Plugins:process_auto_fields(tbl, "insert")
  if not p then
    return nil, err
  end
```
same thing on the other call to `Plugins:process_auto_fields`, looking "inside" a return value without checking it first is very risky.